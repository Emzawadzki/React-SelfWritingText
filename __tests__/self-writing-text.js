import React from 'react';
import { shallow } from 'enzyme';
import Swt from '../src/components/SelfWritingText.jsx';

describe('Rendering tests:', () => {
  it('Renders nothing without required props', () => {
    const wrapper = shallow(<Swt />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders correctly with props', () => {
    const wrapper = shallow(<Swt class={'myclass'} textMajorIn={'testingText'} textIn={['testText1', 'testText2', 'testText3']}/>);
    expect(wrapper).toMatchSnapshot();
  });
});