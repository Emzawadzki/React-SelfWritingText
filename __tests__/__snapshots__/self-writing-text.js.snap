// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering tests: Renders correctly with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Swt
    class="myclass"
    pendingTime={1000}
    textIn={
      Array [
        "testText1",
        "testText2",
        "testText3",
      ]
    }
    textMajorIn="testingText"
    willFreeze={true}
    writingInterval={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="myclass__line"
        >
          
          <span
            className="myclass__v-bar"
          >
            |
          </span>
        </p>,
        <p
          className="myclass__line"
        >
          
        </p>,
      ],
      "className": "myclass",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            <span
              className="myclass__v-bar"
            >
              |
            </span>,
          ],
          "className": "myclass__line",
        },
        "ref": null,
        "rendered": Array [
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "|",
              "className": "myclass__v-bar",
            },
            "ref": null,
            "rendered": "|",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "",
            false,
          ],
          "className": "myclass__line",
        },
        "ref": null,
        "rendered": Array [
          "",
          false,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="myclass__line"
          >
            
            <span
              className="myclass__v-bar"
            >
              |
            </span>
          </p>,
          <p
            className="myclass__line"
          >
            
          </p>,
        ],
        "className": "myclass",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              <span
                className="myclass__v-bar"
              >
                |
              </span>,
            ],
            "className": "myclass__line",
          },
          "ref": null,
          "rendered": Array [
            "",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "|",
                "className": "myclass__v-bar",
              },
              "ref": null,
              "rendered": "|",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "",
              false,
            ],
            "className": "myclass__line",
          },
          "ref": null,
          "rendered": Array [
            "",
            false,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Rendering tests: Renders nothing without required props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Swt
    class="self-writing-text"
    pendingTime={1000}
    willFreeze={true}
    writingInterval={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
